src/prefect/engine.py

```diff
@@ -907,7 +904,7 @@ async def orchestrate_flow_run(
                 extra={"send_to_api": False},
             )
 
-        if not state.is_final():
+        if not state.is_final() and not state.is_paused():
```

Comment:
Should we update the `is_final` method to know that `paused` isn't final?

Reply:
`is_final` does return `False` for `Paused`:

```python3
from prefect.states import Paused

Paused().is_final() # False
```

I think the logic here just looks a little confusing at first glance